Mathlib/Algebra/Group/Action/Equidecomp.lean

```diff
@@ -0,0 +1,250 @@
+/-
+Copyright (c) 2024 Felix Weilacher. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Felix Weilacher
+-/
+import Mathlib.Algebra.Group.Action.Basic
+import Mathlib.Algebra.Group.Pointwise.Finset.Basic
+import Mathlib.Logic.Equiv.PartialEquiv
+
+/-!
+# Equidecompositions
+
+This file develops the basic theory of equidecompositions.
+
+## Main Definitions
+
+Let `G` be a group acting on a space `X`, and `A B : Set X`.
+
+* An *Equidecomposition* of `A` and `B` is typically defined as a finite partition of `A` together
+  with a finite list of elements of `G` of the same size such that applying each element to the
+  matching piece of the partition yields a partition of `B`.
+
+  This yields a bijection `f : A ≃ B` where, given `a : A`, `f a = γ • a` for `γ : G` the group
+  element for `a`'s piece of the partition. Reversing this is easy, and so we get an equivalent
+  (up to the choice of group elements) definition: an *Equidecomposition* of `A` and `B` is a
+  bijection `f : A ≃ B` such that for some `S : Finset G`, `f a ∈ S • a` for all `a`.
+
+  We take this as our definition as it is easier to work with. It is implemented as an element
+  `PartialEquiv X X` with source `A` and target `B`.
+
+## Implementation Notes
+
+* Equidecompositions are implemented as elements of `PartialEquiv X X` together with a
+  `Finset` of elements of the acting group and a proof that every point in the source is moved
+  by an element in the finset.
+
+* The requirement that `G` be a group is relaxed where possible.
+
+* We introduce a non-standard predicate, `DecompOn`, to state that a function satisfies the main
+  combinatorial property of equidecompositions, even if it is not injective or surjective.
+
+## TODO
+
+* Prove that if two sets embeddidecompose into eachother, they are equidecomposable
+  (Schroeder-Bernstein type theorem)
+* Define embeddidecomposability as a Preorder on sets and prove that its induced equivalence
+  relation is equidecomposability.
+* Prove the definition of equidecomposition used here is equivalent to the more familiar one
+  using partitions.
+
+-/
+
+variable {X G : Type*} {A B C: Set X}
+
+open Function Set Pointwise PartialEquiv
+
+namespace Equidecomp
+
+section SMul
+
+variable [SMul G X]
+
+--Should this be private?
+/-- Let `G` act on a space `X` and `A : Set X`. We say `f : X → X` is a `DecompOn` `A`
```

Comment:
Since it's been decided, you can drop the private comment.

```suggestion
variable [SMul G X]

/-- Let `G` act on a space `X` and `A : Set X`. We say `f : X → X` is a `DecompOn` `A`
```